org.omg.sysml.xtext/src/org/omg/sysml/xtext/SysML.xtext

```diff
@@ -2173,348 +2092,17 @@ RenderingRefUsage returns SysML::RenderingUsage :
-		operator = '@[' operand_comp += OwnedExpression ']' )?		
-;
-
-// Unary Expressions
-
-UnaryExpression returns SysML::Expression:
-	  {SysML::OperatorExpression} operator = UnaryOperator operand_comp += ExtentExpression
-	| ExtentExpression
-;
-
-UnaryOperator :
-	'+' | '-' | '!' | '~'
-;
-
-// Extent Expressions
-
-ExtentExpression returns SysML::Expression :
-	  {SysML::OperatorExpression} operator = 'all' ownedFeatureMembership_comp += TypeReferenceMember
-	| SequenceExpression
-;
-
-/* Sequence Expressions */
-
-SequenceExpression returns SysML::Expression :
-	( SequenceConstructionExpression | BaseExpression )
-	( {SysML::OperatorExpression.operand_comp += current} 
-	  operator = '[' operand_comp += OwnedExpression ']'
-	| {SysML::OperatorExpression.operand_comp += current} '->' 
-	  operator = Name ( ownedFeatureMembership_comp += BodyExpressionMember )+
-	)*
-;
-
-BodyExpressionMember returns SysML::FeatureMembership :
-	ownedMemberFeature_comp = BodyExpression
-;
-
-BodyExpression returns SysML::BlockExpression :
-	   => ( ownedFeatureMembership_comp += BodyParameterMember 
-	   	    ( ownedFeatureMembership_comp += BodyParameterMember )* 
-	        '(' ownedFeatureMembership_comp += ResultExpressionMember ')' )
-	 | ownedRelationship_comp += ExpressionTyping
-;
-
-BodyParameterMember returns SysML::ParameterMembership :
-	memberName = Name ownedMemberParameter_comp = BodyParameter
-;
-
-BodyParameter returns SysML::Usage :
-	{SysML::ReferenceUsage} ( TypedBy MultiplicityPart? | MultiplicityPart TypedBy? )?
-;
-
-ExpressionTyping returns SysML::FeatureTyping :
-	type = [ SysML::Function | QualifiedName ]
-;
-
-SequenceConstructionExpression returns SysML::Expression :
-	  {SysML::NullExpression} '{' '}'
-	| '{' OwnedExpression
-	      ( {SysML::OperatorExpression.operand_comp += current} operator = ',' operand_comp += SequenceElementList
-	      | {SysML::OperatorExpression.operand_comp += current} operator = '..' operand_comp += OwnedExpression
-	      )?
-	  '}'
-;
-
-SequenceElementList returns SysML::Expression :	
-	OwnedExpression ( {SysML::OperatorExpression.operand_comp += current} operator = ',' operand_comp += SequenceElementList )?
-;
-
-/* Base Expressions */
-
-BaseExpression returns SysML::Expression :
-	  NullExpression
-	| LiteralExpression 
-	| FeatureReferenceExpression 
-	| InvocationExpression 
-	| ExtentExpression_Deprecated
-	| '(' OwnedExpression ')'
-;
-
-// Null Expressions
-
-NullExpression returns SysML::NullExpression :
-	{SysML::NullExpression} 'null'
-;
-
-// Feature Reference Expressions
-
-FeatureReferenceExpression returns SysML::FeatureReferenceExpression :
-	ownedFeatureMembership_comp += FeatureReferenceMember
-;
-
-FeatureReferenceMember returns SysML::ReturnParameterMembership :
-	ownedMemberFeature_comp = FeatureReference
+	 KerMLExpressions::OwnedExpression | QueryPathExpression
 ;
 
 // Note: Use subsetting here in order to inherit typeing of referent.
+@Override 
```

Comment:
The instance in KerML was returning a SysML::FeatureReferenceExpression instead of ReferenceUsage - if the return types can be unified, it is possible to remove this overridden rule from sysml.xtext.

Reply:
Actually, in KerML it was returning a SysML::Feature. ReferenceUsage is a kind of Feature. Returning just a plain Feature is fine in the SysML grammar, so the override is not needed.